### ISSUE TITLE
Don't explode on tag mismatches

### ISSUE THREAD

**Issue by aroman** [fb55/htmlparser2#19]
**tl;dr** don't do [this](https://github.com/FB55/node-htmlparser/blob/master/lib/DomHandler.js#L39)

Okay first off, let me just say that htmlparser2 is the bomb. I throw some pretty disgusting HTML at it and it chews through it like a boss.

Except for one problem. Sometimes I throw REALLY ugly HTML at it. Specifically, imagine someone took a reasonably valid html page and threw an opening comment ("<!--") somewhere around 90% down the source, and then never closed it. Overwriting otherwise valid HTML content. Sure, the page wouldn't render properly (mainly the footer content just gets lost), but browsers still render most of the page.

htmlparser2 just breaks. Specifically, it breaks on like 39: https://github.com/FB55/node-htmlparser/blob/master/lib/DomHandler.js#L39

The parser should just call it a day and wrap up the DOM tree up to that point instead of exploding. Otherwise my whole server crashes with an exception and I don't get ANY parsed data back. (Yes, I know I could work around the crashing part.)

So, please, could we not throw an exception there and instead die gracefully, albeit prematurely? (Y'know, like browsers do?)

Thanks!
